### PR TITLE
Implements basic Caffe model

### PR DESCRIPTION
**Pull request by GisliGeorgs**
* Currently hardcoded to use self-developed cup-detecting model
* Changed ML implementations to use opencv matrices instead of frame(depth or rgb)/frameset
* Basic implementation of Caffe model

### CODE REVIEW

File: Sight_++_Solution/Intel_Sight_PP/src/classification_result.hpp [hunk 1]
```diff
@@ -6,16 +6,18 @@
...
 	double y;
+
+	point(const int x, const int y) : x(x), y(y) {}
 };
 
 struct ClassificationItem
 {
 	std::string name;
 	double distance;
-	point topLeft;
```
Comment (SvenStruanFinlay) [inline Sight_++_Solution/Intel_Sight_PP/src/classification_result.hpp:15]: Rectangle needs to be the bottom right top left as (0,0) coordinate is top left corner.

**Review by SvenStruanFinlay — CHANGES_REQUESTED** [review]
Sight_++_Solution/Intel_Sight_PP/src/classification_result.hpp

Rectangle needs to be the bottom right top left as (0,0) coordinate is top left corner.

File: Sight_++_Solution/Intel_Sight_PP/src/classification_result.hpp (hunk 1 above)
Comment (GisliGeorgs) [reply Sight_++_Solution/Intel_Sight_PP/src/classification_result.hpp:15]: Are you sure on the coordinate start point?
This is how it is done in the [rs-dnn example](https://github.com/IntelRealSense/librealsense/blob/master/wrappers/opencv/dnn/rs-dnn.cpp).

Comment (SvenStruanFinlay) [reply Sight_++_Solution/Intel_Sight_PP/src/classification_result.hpp:15]: Eh, lets just see what happens then since they do it the normal way

**Review by SvenStruanFinlay — COMMENTED** [review]
Seems good

**Pull request merged**